### PR TITLE
postgresql_ext: cascade extension creating/deleting

### PR DESCRIPTION
**Pull request by kostiantyn-nemchenko**
##### SUMMARY

Currently, without CASCADE clause, if an extension has an unfulfilled dependency on another extension,
CREATE EXTENSION errors out with "required extension ... is not installed" or, if an extension is a dependency for another extension, DROP EXTENSION fails with "cannot drop extension ... because other objects depend on it".
This PR adds support for creating/deleting extensions with their dependencies.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

postgresql_ext
##### ADDITIONAL INFORMATION

Task example:
```
- postgresql_ext:
    name: earthdistance
    db: template1
    cascade: true
```

### CODE REVIEW

File: lib/ansible/modules/database/postgresql/postgresql_ext.py [hunk 1]
```diff
@@ -114,18 +121,26 @@ def ext_exists(cursor, ext):
     return cursor.rowcount == 1
 
 
-def ext_delete(cursor, ext):
+def ext_delete(cursor, ext, cascade):
     if ext_exists(cursor, ext):
-        query = "DROP EXTENSION \"%s\"" % ext
+        if cascade:
```
Comment (andytom) [inline lib/ansible/modules/database/postgresql/postgresql_ext.py:138]: Would it be better to change the logic here to be:
```python
        query = "DROP EXTENSION \"%s\"" % ext
        if cascade:
            query += " CASCADE"
```

File: lib/ansible/modules/database/postgresql/postgresql_ext.py [hunk 2]
```diff
@@ -114,18 +121,26 @@ def ext_exists(cursor, ext):
...
         return True
     else:
         return False
 
 
-def ext_create(cursor, ext):
+def ext_create(cursor, ext, cascade):
     if not ext_exists(cursor, ext):
-        query = 'CREATE EXTENSION "%s"' % ext
+        if cascade:
```
Comment (andytom) [inline lib/ansible/modules/database/postgresql/postgresql_ext.py:151]: Would it be better to change the logic here to be:
```python
        query = "CREATE EXTENSION \"%s\"" % ext
        if cascade:
            query += " CASCADE"
```

File: lib/ansible/modules/database/postgresql/postgresql_ext.py (hunk 1 above)
Comment (kostiantyn-nemchenko) [reply lib/ansible/modules/database/postgresql/postgresql_ext.py:138]: Thank you. It's fixed now

File: lib/ansible/modules/database/postgresql/postgresql_ext.py (hunk 2 above)
Comment (kostiantyn-nemchenko) [reply lib/ansible/modules/database/postgresql/postgresql_ext.py:151]: Thank you. It's fixed now

**Comment by kostiantyn-nemchenko** [comment]
ready_for_review

File: lib/ansible/modules/database/postgresql/postgresql_ext.py [hunk 3]
```diff
@@ -147,6 +158,7 @@ def main():
             db=dict(required=True),
             ext=dict(required=True, aliases=['name']),
             state=dict(default="present", choices=["absent", "present"]),
+            cascade=dict(type='bool', default='no'),
```
Comment (andytom) [inline lib/ansible/modules/database/postgresql/postgresql_ext.py]: Sorry didn't notice this first time around but I think should this be a boolean not a string.
```suggestion
            cascade=dict(type='bool', default=False),
```

Comment (kostiantyn-nemchenko) [reply lib/ansible/modules/database/postgresql/postgresql_ext.py]: Indeed. Thank you

**Comment by andytom** [comment]
shipit

**Comment by andytom** [comment]
bot_status

**Comment by kostiantyn-nemchenko** [comment]
bot_status

File: lib/ansible/modules/database/postgresql/postgresql_ext.py [hunk 4]
```diff
@@ -67,6 +67,13 @@
       - The database extension state
     default: present
     choices: [ "present", "absent" ]
+  cascade:
+    description:
+      - Automatically install/remove any extensions that this extension depends on
+        that are not already installed/removed.
+    type: bool
+    default: no
+    version_added: '2.8'
```
Comment (Andersson007) [inline lib/ansible/modules/database/postgresql/postgresql_ext.py:80]: Add a couple of examples for "present" and "absent" state with "cascade: true" to EXAMPLES block

**Review by Andersson007 — CHANGES_REQUESTED** [review]

**Comment by Andersson007** [comment]
Also add information to 'notes' section that 'cascade' parameter is supported from version 9.6 of PgSQL.

I've tried it, it works correctly (absent, present).

**Comment by kostiantyn-nemchenko** [comment]
Resolved merge conflicts and addressed @Andersson007's feedback.

**Comment by kostiantyn-nemchenko** [comment]
ready_for_review

**Comment by Andersson007** [comment]
I tested it again, just in case. Can be really useful, thank you!

shipit

**Comment by Andersson007** [comment]
shipit

**Review by Andersson007 — APPROVED** [review]

**Comment by kostiantyn-nemchenko** [comment]
bot_status

**Pull request merged**